### ISSUE TITLE
Incompatible with h5py 3.3.0

### ISSUE THREAD

**Issue by peggjt** [pyscf/pyscf#1016]
PySCF is incompatible with h5py-3.3.0 (but seems to work with h5py-3.2.1).

The error is:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ec2-user/.local/lib/python3.7/site-packages/pyscf/__init__.py", line 71, in <module>
    from pyscf import lib
  File "/home/ec2-user/.local/lib/python3.7/site-packages/pyscf/lib/__init__.py", line 24, in <module>
    from pyscf.lib import numpy_helper
  File "/home/ec2-user/.local/lib/python3.7/site-packages/pyscf/lib/numpy_helper.py", line 27, in <module>
    from pyscf.lib import misc
  File "/home/ec2-user/.local/lib/python3.7/site-packages/pyscf/lib/misc.py", line 46, in <module>
    h5py.get_config().default_file_mode = 'a'
  File "h5py/h5.pyx", line 179, in h5py.h5.H5PYConfig.default_file_mode.__set__
ValueError: Using default_file_mode other than 'r' is no longer supported. Pass the mode to h5py.File() instead.
```

**Comment by verena-neufeld** [comment]
Is this similar to issue #1003 ?

**Comment by peggjt** [comment]
> Is this similar to issue #1003 ?

Hi @verena-neufeld,
Thank you. In this case I think the issue is more specific to 3.3.0 versioning, where it might not be working as it is using 'a' mode for accessing file, which is not supported anymore. An instance of this being an issue is after installation on an AWS image, PySCF wont import correctly.

**Comment by fishjojo** [comment]
This is the same issue as described in #1003.
The problem has been fixed in PySCF v2.0, and it will be released in the near future. 
Currently, you may install PySCF manually. See instructions here https://pyscf.org/install.html#compiling-from-source-code
Or you could try removing the lines which cause the problem. https://github.com/pyscf/pyscf/issues/1003#issuecomment-871391527
